fluster/system_info.py

```diff
@@ -0,0 +1,401 @@
+import subprocess
+from typing import Any, Dict, List, Optional
+
+
+class SystemInfo:
+    """Collects hardware and software information about the system"""
+
+    def __init__(self) -> None:
+        self.os_name = platform.system()
+        self.os_version = platform.release()
+        self.cpu_model = self._get_cpu_model()
+        self.gpu_info = self._get_gpu_info()
+        self.total_ram = self._get_total_ram()
+        self.backend_info = self._get_backend_info()
+
+    @staticmethod
+    def _run_command(command: List[str]) -> str:
+        """Run a command and return its output, or empty string on failure"""
+        try:
+            result = subprocess.run(command, capture_output=True, text=True, check=False)
+            if result.returncode == 0:
+                return result.stdout.strip()
+        except (FileNotFoundError, OSError):
+            pass
+        return ""
+
+    @staticmethod
+    def _parse_wmic_list(output: str, key: str) -> Optional[str]:
+        """Parse wmic /format:list output for a specific key"""
+        for line in output.split("\n"):
+            if line.startswith(f"{key}="):
+                value = line.split("=", 1)[1].strip()
+                if value:
+                    return value
+        return None
+
+    @staticmethod
+    def _parse_wmic_table(output: str, skip_header: str) -> Optional[str]:
+        """Parse wmic table format, skipping header and empty lines"""
+        lines = [item.strip() for item in output.split("\n") if item.strip()]
+        for line in lines[1:] if len(lines) > 1 else []:
+            if line and line != skip_header:
+                return line
+        return None
+
+    @staticmethod
+    def _bytes_to_gb(bytes_value: int) -> str:
+        """Convert bytes to GB with one decimal place"""
+        gb = bytes_value / (1024 * 1024 * 1024)
+        return f"{gb:.1f} GB"
+
+    def _get_cpu_model(self) -> str:
+        """Get CPU model information"""
+        try:
+            if self.os_name == "Linux":
+                with open("/proc/cpuinfo", encoding="utf-8") as f:
+                    for line in f:
+                        if "model name" in line:
+                            return line.split(":")[1].strip()
+            elif self.os_name == "Darwin":
+                output = self._run_command(["sysctl", "-n", "machdep.cpu.brand_string"])
+                if output:
+                    return output
+            elif self.os_name == "Windows":
+                output = self._run_command(["wmic", "cpu", "get", "name", "/format:list"])
+                if output:
+                    cpu_name = self._parse_wmic_list(output, "Name")
+                    if cpu_name:
+                        return cpu_name
+
+                output = self._run_command(["wmic", "cpu", "get", "name"])
+                if output:
+                    cpu_name = self._parse_wmic_table(output, "Name")
+                    if cpu_name:
+                        return cpu_name
```

Comment:
do we need 2 approaches for Windows?

If yes and they are mutually exclusive i would suggest to have them in the same conditional statement
```suggestion
                output = self._run_command(["wmic", "cpu", "get", "name", "/format:list"])
                if output:
                    cpu_name = self._parse_wmic_list(output, "Name")
                else:
                    output = self._run_command(["wmic", "cpu", "get", "name"])
                    if output:
                         cpu_name = self._parse_wmic_table(output, "Name")

                 if cpu_name:
                     return cpu_name
```